the-collab-lab/tcl-70-smart-shopping-list
# Issue #20 | Refactor list page using Material UI and following wireframe.

## Description


This code uses components from Material UI to recreate the wireframe layout for BOTH browser and mobile. Although the issue only asks to address browser view only, it was easy enough to make the Material UI components responsive for BOTH browser and mobile.

Wireframe for browser:

![image](https://github.com/the-collab-lab/tcl-70-smart-shopping-list/assets/28488515/b5313ce6-0ac5-4f63-bc68-f488f5e05b18)


Wireframe for mobile:

![image](https://github.com/the-collab-lab/tcl-70-smart-shopping-list/assets/28488515/95d55005-b5c5-40ff-a6ed-1344df0460bc)


## Related Issue

closes #44 

## Acceptance Criteria

- [x] Search bar at the top should still be functional. A material UI text field can be used to replace the search input (https://mui.com/material-ui/react-text-field/#icons)
- [x] The 'Add list item" form should also be functional. Once again material UI text fields can be used to replace form inputs (https://mui.com/material-ui/react-text-field/#form-props)
- [x] Each list item should be placed in their own material UI card (https://mui.com/material-ui/react-card/)
- [x] All of the functionality for each list item (the purchased checkbox, delete button) should work as normal


## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|  ✓    | :sparkles: New feature     |
|  ✓    |:sparkles: Enhancement |

## Updates

## Before
- Before Browser View

![image](https://github.com/the-collab-lab/tcl-70-smart-shopping-list/assets/28488515/cd572c18-1a35-4fb3-b091-ea6c55b3cfe7)

- Before Mobile View

![image](https://github.com/the-collab-lab/tcl-70-smart-shopping-list/assets/28488515/fbf754f5-db93-458b-89ee-30c5a89c08c3)




### After

- After Browser View
 
![image](https://github.com/the-collab-lab/tcl-70-smart-shopping-list/assets/28488515/043e8a6d-4450-4398-ba11-1b3255b1c863)

- After Mobile View

![image](https://github.com/the-collab-lab/tcl-70-smart-shopping-list/assets/28488515/f64cad93-d02e-4778-a94d-7d27765d352e)

## Testing Steps / QA Criteria

* From your terminal, pull down this branch with `git pull origin gl-km-wireframe-list-page` and check that branch out with `git checkout gl-km-wireframe-list-page` 
* In the terminal, enter `npm start`
* In the browser, go to localhost:3000 and select a list to navigate to the list page and view items in the list.
* On the list page try : searching for an item, adding a new item and deleting an item.
* Using your browsers developer tools, test the responsiveness of the page and test all the features as before (search items, add items, delete items)
